### ISSUE TITLE
Label column is required again in static pipe

### ISSUE THREAD

**Issue by Zruty0** [dotnet/machinelearning#1047]
We didn't add any provision to make certain estimators 'train-only'.

Compare the following:
```c#
var pipeline = new ConcatEstimator(env, "Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth")
    .Append(new TermEstimator(env, "Label"), TransformerScope.TrainTest)
    .Append(new SdcaMultiClassTrainer(env, new SdcaMultiClassTrainer.Arguments { MaxIterations = 100, Shuffle = true, NumThreads = 1 }, "Features", "Label"))
    .Append(new KeyToValueEstimator(env, "PredictedLabel"));

var model = pipeline.Fit(data).GetModelFor(TransformerScope.Scoring); // <-- that's how we can clean up the pipeline
var engine = model.MakePredictionFunction<IrisDataNoLabel, IrisPrediction>(env);
```
And here is the static one:
```c#
var learningPipeline = reader.MakeNewEstimator()
    .Append(r => (
        r.Label,
        Features: r.SepalLength.ConcatWith(r.SepalWidth, r.PetalLength, r.PetalWidth)))
    .Append(r => (
        r.Label,
        Predictions: classification.Trainers.Sdca(r.Label.ToKey(), r.Features)))
    .Append(r => r.Predictions.predictedLabel.ToValue());
```
Can we keep the concise pipeline description, yet allow for something like 'GetModelForScoring' in some way?

/cc @TomFinley  @markusweimer

**Comment by codemzs** [comment]
static pipe API is no longer being developed.
